packages/@sanity/form-builder/src/inputs/ReferenceInput/ArrayItemReferenceInput.tsx

```diff
@@ -463,13 +476,12 @@ export const ArrayItemReferenceInput = forwardRef(function ReferenceInput(
           <Box flex={1}>
             <Flex align="center">
               {hasRef ? (
-                <Card
+                <OptionCard
                   flex={1}
                   padding={1}
                   paddingRight={3}
                   radius={2}
-                  as={EditReferenceLink}
-                  //@ts-expect-error issue with styled components "as" polymorphism
+                  forwardedAs={EditReferenceLink as any}
```

Comment:
Would be interesting to understand what the issue is with typescript here. Any ideas?